### PR TITLE
Add --cache option to lint and prettier scripts

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

Adds the `--cache` and `--cache-strategy content` options to eslint and prettier scripts

### Why is it needed?

Linting currently takes a very long time (~50s on my local environment on `main`, ~80s when adding typescript linting in `features/deits`).

This change reduces subsequent runs after the first to a tiny fraction of the full run and should save developers about a minute of waiting before every strapi commit, since we run prettier and `lint --fix` with a commit hook.

### How to test it?

Run `yarn lint` and it should be cached and run faster on subsequent runs. When you update a file, it should be linted again the next time it's run.

The important thing is that it does not accidentally use the cache for something that should be re-checked. For that reason, I opted to use cache-strategy content in case switching between git branches could update content without updating a timestamp.

### CODE REVIEW

**Review by Convly — COMMENTED** [review]
Looks like a nice improvement time-wise. I went from 60sc to 3sc lint.
Do you know how it'll play with the CI environments + where is the cache stored?

**Comment by innerdvations** [comment]
The prettier cache is stored in the node_modules/prettier folder

The eslint cache is stored in the strapi root as .eslintcache, which is why I added that to .gitignore. In theory it should persist between branch changes (and only re-lint those files that have changed), although from my testing it generally seemed to invalidate the entire cache each switch anyway.

**Review by gu-stav — COMMENTED** [review]
Overall this looks very good to me. What do you think about moving the eslint cache into `node_modules` too? We restore the whole folder from the cache in GH Actions, and they could therefore benefit from this change too.

```
--cache-location ./node_modules/.cache/eslint
```

**Comment by innerdvations** [comment]
> Overall this looks very good to me. What do you think about moving the eslint cache into `node_modules` too? We restore the whole folder from the cache in GH Actions, and they could therefore benefit from this change too.
> 
> ```
> --cache-location ./node_modules/.cache/eslint
> ```

Ooh, I had no idea. That would be great if we could save a few minutes on the GH workflow too.

File: .gitignore [hunk 1]
```diff
@@ -80,6 +80,7 @@ $RECYCLE.BIN/
 .idea
 nbproject
 .vscode/
+.eslintcache
```
Comment (innerdvations) [inline .gitignore:83]: This isn't needed anymore, but I'll leave it because it's nice to have in case someone runs eslint --cache manually

**Review by Convly — APPROVED** [review]

**Pull request merged**